client/state/invites/reducer.js

```diff
@@ -0,0 +1,55 @@
+/** @format */
+
+/**
+ * Internal dependencies
+ */
+import { combineReducers, createReducer } from 'state/utils';
+import {
+	INVITES_REQUEST,
+	INVITES_REQUEST_FAILURE,
+	INVITES_REQUEST_SUCCESS,
+} from 'state/action-types';
+
+/**
+ * Returns the updated site invites requests state after an action has been
+ * dispatched. The state reflects a mapping of site ID to a boolean reflecting
+ * whether a request for the post is in progress.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function requesting( state = {}, action ) {
+	switch ( action.type ) {
+		case INVITES_REQUEST:
+		case INVITES_REQUEST_SUCCESS:
+		case INVITES_REQUEST_FAILURE:
+			return Object.assign( {}, state, {
+				[ action.siteId ]: INVITES_REQUEST === action.type,
+			} );
+	}
+
+	return state;
+}
+
+/**
+ * Tracks all known invite objects as an object indexed by site ID and
+ * containing arrays of invites.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export const items = createReducer(
+	{},
+	{
+		[ INVITES_REQUEST_SUCCESS ]: ( state, action ) => {
+			return {
+				...state,
+				[ action.siteId ]: action.invites,
```

Comment:
For brevity (avoid storing unnecessary data in state) and clarity (make it possible to tell from within Calypso what properties an invite object will have), we should build an object containing only specific properties here, like we do for [post likes](https://github.com/Automattic/wp-calypso/blob/3441d433/client/state/posts/likes/reducer.js#L69-L71).  I haven't done this yet because it will need to change later, but there wouldn't be any harm in including it in this PR too.